docs/DeveloperGuide.md

```diff
@@ -1,38 +1,1061 @@
+
+Each endpoint is a child class `Endpoint`. Currently, there are 2 endpoints available:
+
+| Endpoint   | Method to call          |
+| ---------- | ----------------------- |
+| `/entry`   | `callEntryEndpoint()`   |
+| `/entries` | `callEntriesEndpoint()` |
+
+##### Creating a request
+
+- To create a request, simply instantiate `pocketpal.communication.pocketpal.Request`
+  with the desired request method.
+- If there are any parameters associated with the request, you may add them using `addParam()`
+
+```java
+Request req = new Request(RequestMethod.PATCH);
+req.addParam(RequestParams.EDIT_DESCRIPTION, "mango juice");
+```
+
+##### Making a request
+
+- To call each endpoint, pass the `Request` into its corresponding method in the frontend.
+- A `Response` will be returned to the frontend.
+
+> All request body and parameter data should be serialised with `String.valueOf()` if not specified.
+
+```java
+Backend backend = new Backend();
+Response res = backend.callEntryEndpoint(req);
+
+if (res.getResponseStatus() != ResponseStatus.OK) {
+   // handle status        
+}
+
+Entry entry = EntryParser.deserialise(res.getData());
+// process entry
+```
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+
+#### Access all entries available
+##### Get recent or all entries
+
+<details>
+   <summary>Sequence diagram</summary>
+
+   <img alt="Entries Endpoint [GET] Sequence Diagram" src="docs/../static/backend/EntriesEndpointGetSequence.png" />
+
+</details>
+
+`GET /entries`
+
+__Body__
+
+Number of recent entries to view if present, otherwise all entries will be returned.
+
+__Parameters__
+
+__`GET_SIZE`__ boolean
+
+- If present, response will contain the number of entries.
+- Otherwise, the following filters will be applied to the entries:
+
+__`FILTER_BY_AMOUNT_START`__ double
+
+- If this is the only parameter, it will be used as a minimum amount filter.
+- If both filter amount parameters are present, it will be used as a amount range filter.
+
+__`FILTER_BY_AMOUNT_END`__ double
+
+- If this is the only parameter, it will be used as a maximum amount filter.
+- If both filter amount parameters are present, it will be used as a amount range filter.
+
+__`FILTER_BY_CATEGORY`__ Category
+
+- Filter entries by category
+
+__`FILTER_BY_QUERY`__ String
+
+- Filter entries by user query
+
+__Responses__
+
+| Status Code | Description           | Remarks                                                                       |
+| ----------- | --------------------- | ----------------------------------------------------------------------------- |
+| `200`       | OK                    | Gson-serialised `List<Entry>`, deserialise with `EntryLogParser::deserialise` |
+| `422`       | Unprocessable Content | -                                                                             |
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
```

Comment:
Good job including these links, they were very useful in navigating your document!